smarts/core/sumo_traffic_simulation.py

```diff
@@ -680,11 +687,11 @@ def remove_traffic_vehicle(self, vehicle_id: str):
         self._traci_conn.vehicle.remove(vehicle_id)
         self._sumo_vehicle_ids.remove(vehicle_id)
 
-    def _shape_of_vehicle(self, vehicle_id):
-        p = self._traci_conn.vehicle.getPosition(vehicle_id)
-        length = self._traci_conn.vehicle.getLength(vehicle_id)
-        width = self._traci_conn.vehicle.getWidth(vehicle_id)
-        heading = Heading.from_sumo(self._traci_conn.vehicle.getAngle(vehicle_id))
+    def _shape_of_vehicle(self, sumo_vehicle_state_all, vehicle_id):
+        p = sumo_vehicle_state_all[vehicle_id][66]  # getPosition
+        length = sumo_vehicle_state_all[vehicle_id][68]  # getLength
+        width = sumo_vehicle_state_all[vehicle_id][77]  # getWidth
+        heading = Heading.from_sumo(sumo_vehicle_state_all[vehicle_id][67])  # getAngle
```

Comment:
Please use `tc.VAR_POSITION`,  `tc.VAR_LENGTH`, `tc.VAR_WIDTH`, and `tc.VAR_ANGLE`.

Reply:
A structure's variables are now referred to by their pre-defined names.